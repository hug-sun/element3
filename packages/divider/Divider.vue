<template>
  <div v-bind="$attrs" :class="['el-divider', `el-divider--${direction}`]">
    <div
      v-if="$slots.default && direction !== 'vertical'"
      :class="['el-divider__text', `is-${contentPosition}`]"
    >
      <slot />
    </div>
  </div>
</template>

<script>
export default {
  name: 'ElDivider',
  props: {
    direction: {
      type: String,
      default: 'horizontal',
      validator(val) {
        return ['horizontal', 'vertical'].indexOf(val) !== -1
      }
    },
    contentPosition: {
      type: String,
      default: 'center',
      validator(val) {
        return ['left', 'center', 'right'].indexOf(val) !== -1
      }
    }
  }
}
</script>
