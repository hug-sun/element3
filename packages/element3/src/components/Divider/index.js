import ElDivider from './src/Divider'

/* istanbul ignore next */
ElDivider.install = function (app) {
  app.component(ElDivider.name, ElDivider)
}

export { ElDivider }
