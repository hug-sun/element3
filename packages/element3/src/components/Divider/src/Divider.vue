<template>
  <div v-bind="$attrs" :class="['el-divider', `el-divider--${direction}`]">
    <div
      v-if="$slots.default && direction !== 'vertical'"
      :class="['el-divider__text', `is-${contentPosition}`]"
    >
      <slot />
    </div>
  </div>
</template>

<script>
export default {
  name: 'ElDivider',
  props: {
    direction: {
      type: String,
      default: 'horizontal',
      validator(val) {
        return ['horizontal', 'vertical'].includes(val)
      }
    },
    contentPosition: {
      type: String,
      default: 'center',
      validator(val) {
        return ['left', 'center', 'right'].includes(val)
      }
    }
  }
}
</script>
