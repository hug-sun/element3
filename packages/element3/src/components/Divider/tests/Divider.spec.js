import Divider from '../src/Divider.vue'
import { mount } from '@vue/test-utils'

describe('Divider.vue', () => {
  it('snapshot', () => {
    const wrapper = mount(Divider)
    expect(wrapper.element).toMatchSnapshot()
  })
  it('content', () => {
    const vm = mount(Divider, {
      slots: {
        default: () => 'Im perfect divider!'
      }
    })

    expect(vm.text()).toBe('Im perfect divider!')
  })

  it('direction', () => {
    const vm = mount(Divider, {
      props: { direction: 'vertical' }
    })

    expect(vm.classes()).toContain('el-divider--vertical')
  })

  it('contentPosition', () => {
    const vm = mount(Divider, {
      props: { 'content-position': 'left' },
      slots: {
        default: () => {
          return 'some text'
        }
      }
    })
    const text = vm.find('.el-divider__text')
    expect(text).toBeDefined()
    expect(text.classes()).toContain('is-left')
  })

  it('apply class to divider', () => {
    const vm = mount(Divider, {
      attrs: {
        class: 'my-divider'
      }
    })

    expect(vm.classes()).toContain('my-divider')
  })
})
