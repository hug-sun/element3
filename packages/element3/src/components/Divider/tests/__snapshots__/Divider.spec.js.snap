// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider.vue snapshot 1`] = `
<div
  class="el-divider el-divider--horizontal"
>
  <!--v-if-->
</div>
`;
